src/blop/agent.py

```diff
@@ -293,101 +435,370 @@ def ask(self, acqf="qei", n=1, route=True, sequential=True, upsample=1, **acqf_k
+        self._detector_names = list(names)
+
+    def server_registrations(self):
+        """This is how we make these avaialble to the REST API."""
+        self._register_method("Update Acquistion Function Kwargs", self.update_acqf_kwargs)
+        self._register_property("Acquisition Function", self.acquisition_function, self.acquisition_function)
+        self._register_property("Route Points", self.route, self.route)
+        self._register_property("Sequential Points", self.sequential, self.sequential)
+        self._register_property("Upsample Points", self.upsample, self.upsample)
+        return super().server_registrations()
+
+    def ask(self, batch_size) -> Tuple[Sequence[Dict[str, ArrayLike]], Sequence[ArrayLike]]:
+        default_result = super().ask(
+            n=batch_size,
+            acqf=self._acqf_string,
+            route=self._route,
+            sequential=self._sequential,
+            upsample=self._upsample,
+            **self._acqf_kwargs,
+        )
+
+        """res = {
+            "points": {dof.name: list(points[..., i]) for i, dof in enumerate(active_dofs(read_only=False))},
+            "acqf_name": acqf_config["name"],
+            "acqf_obj": list(np.atleast_1d(acqf_obj.numpy())),
+            "acqf_kwargs": acqf_kwargs,
+            "duration_ms": duration,
+            "sequential": sequential,
+            "upsample": upsample,
+            "read_only_values": read_only_values,
+            # "posterior": p,
+        }
+        """
+
+        points: Dict = default_result.pop("points")
+        acqf_obj: List[ArrayLike] = default_result.pop("acqf_obj")
+        # Turn dict of list of points into list of consistently sized points
+        keys = list(points.keys())
+        points: List[Tuple[ArrayLike]] = list(zip(*[points[key] for key in keys]))
+        dicts = []
+        for point, obj in zip(points, acqf_obj):
+            d = default_result.copy()
+            d["point"] = point
+            d["acqf_obj"] = obj
+            dicts.append(d)
+        return points, dicts
+
+    def tell(self, x, y):
+        x = {key: x_i for x_i, key in zip(x, self.dofs.names)}
+        y = {key: y_i for y_i, key in zip(y, self.objectives.names)}
+        super().tell(data={**x, **y})
+        return {**x, **y}
+
+    def report():
+        raise NotImplementedError("Report is not implmented for BlueskyAdaptiveAgent")
+
+    def unpack_run(self, run):
+        """Use my DOFs to convert the run into an independent array, and my objectives to create the dependent array.
+        In practice for shape management, we will use lists not np.arrays at this stage.
+        Parameters
+        ----------
+        run : BlueskyRun
+
+        Returns
+        -------
+        independent_var :
+            The independent variable of the measurement
+        dependent_var :
+            The measured data, processed for relevance
+        """
+        if self.digestion == default_digestion_function:
```

Comment:
We can remove `default_digestion_function` as a default since it does nothing...
```suggestion
        if not self.digestion:
```

Reply:
```suggestion
        if not self.digestion or self.digestion == default_digestion_function:
```
I'll save that for a future PR since this will default to true in other places doing the simple thing.